src/common/type-check.ts

```diff
@@ -29,5 +29,9 @@ export function isString (value: Result): boolean {
  * @return {boolean} result of the test
  */
 export function isObject (value: unknown): boolean {
```

Comment:
`export function isObject (value: unknown): value is Object {`

Reply:
is that proper TS for return type?
i'm getting:
<img width="1006" alt="type-check_ts______Documents_Development_illogical_src_common__-_VIM2" src="https://user-images.githubusercontent.com/595287/88332269-433b3500-ccfc-11ea-92bc-d5578c40a6dc.png">

Reply:
This is telling ts to assume that value is an `Object`. I've used object here but I should have used something else like the message is mentioning.

`export function isObject (value: unknown): value is Record<string, unknown> {`

Reply:
but why is the return value a record?

Reply:
> export function isObject (value: unknown): value is Object

Yes, is is better, used in the API ... kinda legacy here, housekeeping to update to ```is X``` approach

> Record<string, unknown>

The newer version of TS has a rule for this one, to better use Record<string, unknown>``` rather than ```object```

Reply:
The definition type of a record is
```Typescript
type Record<K, V> = {
  [K]: V
}
```